stream/views/stream.py

```diff
@@ -43,14 +43,17 @@ def get_success_url(self):
         return reverse('stream:lobby_detailview', args=[self.object.lobby.pk])
 
 
-class DeleteView(generic.DeleteView):
-    model = models.Stream
-
-    def delete(self, request, *args, **kwargs):
-        self.object = self.get_object()
-        self.object.removed = True
-        self.object.save()
-        return HttpResponseRedirect(self.get_success_url())
-
-    def get_success_url(self):
-        return reverse('stream:lobby_detailview', args=[self.object.lobby.pk])
+class RemoveView(generic.View):
+
+    def post(self, request, *args, **kwargs):
+        stream = models.Stream.objects.get(pk=kwargs.get('pk'))
+        if (not stream.removed):
```

Comment:
I think you can just remove the else then put the stream.save outside the if statement also the HttpResponse

Reply:
made the changes